src/mon/PGMap.cc

```diff
@@ -2886,7 +2886,7 @@ int reweight::by_utilization(
     for (const auto& pg : pgm.pg_stat) {
       if (pools && pools->count(pg.first.pool()) == 0)
 	continue;
-      for (const auto acting : pg.second.acting) {
+      for (const auto acting : pg.second.up) {
```

Comment:
IMHO, acting set is temporarily response for a PG if encountered some osdmap changes, up set is the destination. we expect PG eventually can be placed in up set, so maybe user will prefer to a final distribution of weight. pg number of a osd act as acting which reweight mechanism relies on will fluctuate during rebalance, the result maybe not that accurate.
anyway, i agree that it might be not an absolute advantage. let it go or keep here?